pkg/actors/actors.go

```diff
@@ -323,38 +323,46 @@ func (a *actorsRuntime) startDeactivationTicker(configuration Config) {
 	}()
 }
 
+type lookupActorRes struct {
+	targetActorAddress string
+	appID              string
+}
+
 func (a *actorsRuntime) Call(ctx context.Context, req *invokev1.InvokeMethodRequest) (*invokev1.InvokeMethodResponse, error) {
 	a.placement.WaitUntilPlacementTableIsReady()
 
 	actor := req.Actor()
-	targetActorAddress, appID := "", ""
 	// Retry here to allow placement table dissemination/rebalancing to happen.
-	var policy resiliency.Runner
+	var policyDef *resiliency.PolicyDefinition
 	if a.isResiliencyEnabled {
-		policy = a.resiliency.BuiltInPolicy(ctx, resiliency.BuiltInActorNotFoundRetries)
+		policyDef = a.resiliency.BuiltInPolicy(resiliency.BuiltInActorNotFoundRetries)
 	} else {
 		noOp := resiliency.NoOp{}
-		policy = noOp.BuiltInPolicy(ctx, resiliency.BuiltInActorNotFoundRetries)
+		policyDef = noOp.BuiltInPolicy(resiliency.BuiltInActorNotFoundRetries)
 	}
-	rErr := policy(func(ctx context.Context) error {
-		targetActorAddress, appID = a.placement.LookupActor(actor.GetActorType(), actor.GetActorId())
-		if targetActorAddress == "" {
-			return errors.Errorf("error finding address for actor type %s with id %s", actor.GetActorType(), actor.GetActorId())
+	policyRunner := resiliency.NewRunner[*lookupActorRes](ctx, policyDef)
+	lar, err := policyRunner(func(ctx context.Context) (*lookupActorRes, error) {
+		rAddr, rAppID := a.placement.LookupActor(actor.GetActorType(), actor.GetActorId())
+		if rAddr == "" {
+			return nil, fmt.Errorf("error finding address for actor type %s with id %s", actor.GetActorType(), actor.GetActorId())
 		}
-		return nil
+		return &lookupActorRes{
+			targetActorAddress: rAddr,
+			appID:              rAppID,
+		}, nil
 	})
-
-	if rErr != nil {
-		return nil, rErr
+	if err != nil {
+		return nil, err
+	}
+	if lar == nil {
```

Comment:
it doesn't really deviate from the previous behavior because before `targetActorAddress` and `appID` were set as a side effect. The policyRunner allows returning one value only, so I had to wrap it in a struct.

Although policyRunner always returns an `*lookupActorRes` object, if the policy method times out then we only get an error and no data value. So `lar` could be nil. In this specific case, if `lar == nil` then automatically `err != nil` so L358 should never be hit....... but an extra nil-check never hurt nobody IMHO